### PR TITLE
Fixing a race condition

### LINKED ISSUE #44: Sometimes iOS simulator does not start in JS debugging mode
I haven't been able to track down a solid repro, but sometimes when launching for the iOS simulator it does not start in debug mode. Once you enable "chrome" debugging, we start to execute the code and hit breakpoints etc.

### PR DESCRIPTION
**Pull request by MSLaguana**
When enabling debug mode on the iOS simulator, we attempt to modify a plist file backing an NSUserDefaults object. This plist file is written to by the app at various times, and this would sometimes overwrite our change.

With this fix, we start the app to ensure the file is created, then we stop the app to avoid a race condition. We need to wait a moment to ensure that the app is completely closed, then we write to the file and restart the app. Simply relaunching the app is not sufficient, we need to re-install the app or it does not pick up the changes.

Fixes #44

### CODE REVIEW

File: src/debugger/ios/iOSPlatform.ts [hunk 1]
```diff
@@ -76,13 +77,36 @@ export class IOSPlatform implements IAppPlatform {
...
+                // Debugging must still be enabled
+                const commandExecutor = new CommandExecutor();
+                const childProcess = new ChildProcess();
+                const launchAppString = `xcrun simctl launch booted ${bundleId}`;
+                // simctl launch returns the process ID of the app in the simulator
+                return childProcess.exec(launchAppString).outcome.then((buffer: Buffer) => {
+                    // Kill the simulated app
+                    const pidMatch = buffer.toString().match(/: ([0-9]+)/);
+                    const pid = pidMatch[1];
+                    return commandExecutor.execute(`kill ${pid} # Restarting app`);
```
Comment (dlebu) [inline src/debugger/ios/iOSPlatform.ts]: Could we use `xcrun simctl spawn booted launchctl stop` here instead? What is the advantage of `kill`?

Comment (MSLaguana) [inline src/debugger/ios/iOSPlatform.ts]: Ah, I was not aware of that command. It seems to work much better and means we do not need the arbitrary delay any more I think. I'll test it some more to be sure. Thanks!

File: src/debugger/ios/iOSPlatform.ts [hunk 2]
```diff
@@ -76,13 +77,38 @@ export class IOSPlatform implements IAppPlatform {
...
+            this.plistBuddy.getBundleId(launchArgs.projectRoot)
+        ]).spread((debugModeSetting: string, bundleId: string) => {
+            if (debugModeSetting !== IOSDebugModeManager.WEBSOCKET_EXECUTOR_NAME) {
+                // Debugging must still be enabled
+                // We enable debugging by writing to a plist file that backs a NSUserDefaults object,
+                // but that file is written to by the app on occasion. To avoid races, we shut the app
+                // down before writing to the file.
+                const childProcess = new ChildProcess();
+
+                return childProcess.exec("xcrun simctl spawn booted launchctl list").outcome.then((buffer: Buffer) => {
```
Comment (digeff) [inline src/debugger/ios/iOSPlatform.ts]: I think that childProcess.execToString is a better match here

**Comment by digeff** [comment]
:+1: with two minor comments. Should we add telemetry to this code?

**Comment by dlebu** [comment]
:+1:

**Pull request merged**